pkg/api/resource_bundle_types.go

```diff
@@ -19,64 +19,67 @@ type ResourceBundleStatus struct {
 }
 
 // DecodeManifestBundle converts a CloudEvent JSONMap representation of a list of resource manifest
-// into manifest bundle payload.
-func DecodeManifestBundle(manifest datatypes.JSONMap) (map[string]any, *workpayload.ManifestBundle, error) {
+// into metadata, a list of manifests, a list of manifest configs, and a delete option for openapi output.
+func DecodeManifestBundle(manifest datatypes.JSONMap) (map[string]any, []map[string]any, []map[string]any, map[string]any, error) {
```

Comment:
nit: would we use a struct {metadata, manifests ... } to represent the returned obj?

Reply:
done.